mica/archive/tests/test_aca_dark_cal.py

```diff
@@ -104,3 +104,29 @@ def test_get_dark_cal_props_table_mixed():
     assert props.colnames == ['ccd_temp', 'date', 'dec', 'dur', 'eb', 'el', 'id', 'l_l0',
                               'ra', 'start', 'stop', 'sun_el', 'zodib', 't_ccd',
                               'n_ccd_img', 'datestart', 'datestop', 'filename']
+
+
+@pytest.mark.skipif('not HAS_DARK_ARCHIVE', reason='Test requires dark archive')
+def test_vectorized():
+    dark_id_ref = ['2022100', '2022105', '2022127']
+    date_ref = ['2022:100', '2022:105', '2022:127']
+    assert np.all(dark_id_ref == dark_cal.date_to_dark_id(date_ref))
+    assert np.all(date_ref == dark_cal.dark_id_to_date(dark_id_ref))
+
+    dark_cal_id_ref = ['2022069', '2022104', '2022104']
+    dark_cal_id = dark_cal.get_dark_cal_id(['2022:100', '2022:105', '2022:127'], 'before')
+    assert np.all(dark_cal_id_ref == dark_cal_id)
+
+    dark_cal_id_ref = ['2022104', '2022133', '2022133']
+    dark_cal_id = dark_cal.get_dark_cal_id(['2022:100', '2022:105', '2022:127'], 'after')
+    assert np.all(dark_cal_id_ref == dark_cal_id)
+
+    dark_cal_id_ref = ['2022104', '2022104', '2022133']
+    dark_cal_id = dark_cal.get_dark_cal_id(['2022:100', '2022:105', '2022:127'], 'nearest')
+    assert np.all(dark_cal_id_ref == dark_cal_id)
+
+
+@pytest.mark.skipif('not HAS_DARK_ARCHIVE', reason='Test requires dark archive')
+def test_lower_limit():
+    with pytest.raises(ValueError, match='trying to get last dark cal on'):
+        dark_cal.get_dark_cal_id('2000:001')
```

Comment:
This seems fine.  I don't know if we'll end up with a use case to put the 1999:223 OAC dark cal in the mica archive more directly.